Tests/SSLPinningTests.m

```diff
@@ -82,11 +87,47 @@ - (void)testSSLPinningInvalidCert {
     }];
 }
 
-- (void)testSSLPinningValidCert {
+- (void)testSSLPinningInvalidCertEU {
     XCTestExpectation *expectation = [self expectationWithDescription:@"Testing Pinning"];
 
     self.amplitude.sslPinningEnabled = YES;
-    [AMPURLSession pinSSLCertificate:@[@"ComodoRsaDomainValidationCA"]];
+    [AMPURLSession sharedSession]; // trigger static instance to pin valid certificates, then pin the invalid one
+    NSDictionary *domainCertFilenamesMap = @{
+        @"test1": @[@"InvalidCertificationAuthority"]
+    };
+    [AMPURLSession pinSSLCertificate:domainCertFilenamesMap];
+
+    [self.amplitude setServerZone:EU];
+    [self.amplitude initializeApiKey:@"361e4558bb359e288ef75d1ae31437a0"];
```

Comment:
@liuyang1520 I believe we handle the server zone first then initialize. How about initializing and then calling the setServerZone method? Do we have a way to support that? cc @yuhao900914

Reply:
The reverse order works too, I use the 1. init, 2. set zone to EU in my local Swift app. It works. I mentioned in the desc above that once `initializeApiKey`, then we cannot init with another API key.

Reply:
I test on both app and test. All get passed.

Reply:
About "after a request is sent", Do you mean they initialized -> logEvent -> setServerZone?

Reply:
I guess this is just about the timing and use case. For the use case I mentioned, the session is actually initialized at https://github.com/amplitude/Amplitude-iOS/blob/main/Sources/Amplitude/Amplitude.m#L1000. Once that is set, the sharedSession is a singleton method and won't update with the certificate again. This is less likely to happen if set up correctly. Just to point out in that case it could break.

Reply:
So if I understand it correctly, my tests below can verify that it should work (with SSL pinning enabled):
1. `Amplitude.instance().initializeApiKey("eu-api-key", userId: "marvin+eu@amplitude.com")`
2. `Amplitude.instance().logEvent("Swift-TEST-SPM:  Button clicked EU 1")`

Events shouldn't show up, as the EU API key doesn't work for the US endpoint, getting this error as expected:
```
2022-07-14 19:01:17.580734-0700 event_emitter[60645:256059] ERROR: Connection response received:400, invalid_api_key
```

In button click action:
4. `Amplitude.instance().setServerZone(AMPServerZone.EU)`
5. `Amplitude.instance().logEvent("Swift-TEST-SPM:  Button clicked EU")`

All events are showing up now: https://analytics.eu.amplitude.com/amplitude/activity, under `marvin-test` project.

I mentioned in the beginning of this PR, the API key is not supported to be updated once init, which means the API key should match the endpoint for sure.
>As we currently don't support reinit API key, so we don't need to support runtime endpoint switch either. This feature is not covered in this PR.

Reply:
@liuyang1520 Interesting, I am more interested in how the certificate gets matched later, when would the certificate update again?

Reply:
The certs are loaded, then matched here: https://github.com/amplitude/Amplitude-iOS/blob/main/Sources/Amplitude/ISPPinnedNSURLSessionDelegate.m#L31. The cert doesn't need to be updated.

Reply:
Just confirmed, android will have the ssl pinning error.

Reply:
Discussed offline with @qingzhuozhen , iOS version already has the cert lazy loading in https://github.com/amplitude/Amplitude-iOS/blob/main/Sources/Amplitude/ISPCertificatePinning.m#L89.